packages/amplify-e2e-tests/src/cleanup-e2e-resources-codebuild.ts

```diff
@@ -306,51 +306,61 @@ const getS3Buckets = async (account: AWSAccountInfo): Promise<S3BucketInfo[]> =>
-  buildBatchArn: _.get(record, ['0', 'cbInfo', 'buildBatchArn']),
-  projectName: _.get(record, ['0', 'cbInfo', 'projectName']),
-  buildComplete: _.get(record, ['0', 'cbInfo', 'buildComplete']),
-  cbJobDetails: _.get(record, ['0', 'cbInfo']),
-  buildStatus: _.get(record, ['0', 'cbInfo', 'buildStatus']),
-});
+const extractCCIJobInfo = (
+  record: S3BucketInfo | StackInfo | AmplifyAppInfo,
+  buildInfos: Record<string, CodeBuild.Build[]>
+  ): CBJobInfo => {
+  const buildId = _.get(record, ['0', 'jobId']);
+  return {
+    buildBatchArn: _.get(buildInfos, [ buildId, '0', 'buildBatchArn' ]),
+    projectName: _.get(buildInfos, [ buildId, '0', 'projectName' ]),
+    buildComplete: _.get(buildInfos, [ buildId, '0', 'buildComplete' ]),
+    cbJobDetails: _.get(buildInfos, [ buildId, '0' ]),
+    buildStatus: _.get(buildInfos, [ buildId, '0', 'buildStatus' ])
+  };
+}
+
 
 /**
  * Merges stale resources and returns a list grouped by the CodeBuild jobId. Amplify Apps that don't have
  * any backend environment are grouped as Orphan apps and apps that have Backend created by different CodeBuild jobs are
  * grouped as MULTI_JOB_APP. Any resource that do not have a CodeBuild job is grouped under UNKNOWN
  */
-const mergeResourcesByCCIJob = (
+const mergeResourcesByCCIJob = async (
   amplifyApp: AmplifyAppInfo[],
   cfnStacks: StackInfo[],
   s3Buckets: S3BucketInfo[],
   orphanS3Buckets: S3BucketInfo[],
   orphanIamRoles: IamRoleInfo[],
-): Record<string, ReportEntry> => {
+): Promise<Record<string, ReportEntry>> => {
   const result: Record<string, ReportEntry> = {};
 
-  const stacksByJobId = _.groupBy(cfnStacks, (stack: StackInfo) => _.get(stack, ['cbInfo', 'id'], UNKNOWN));
+  const stacksByJobId = _.groupBy(cfnStacks, (stack: StackInfo) => _.get(stack, ['jobId'], UNKNOWN));
 
-  const bucketByJobId = _.groupBy(s3Buckets, (bucketInfo: S3BucketInfo) => _.get(bucketInfo, ['cbInfo', 'id'], UNKNOWN));
+  const bucketByJobId = _.groupBy(s3Buckets, (bucketInfo: S3BucketInfo) => _.get(bucketInfo, ['jobId'], UNKNOWN));
 
   const amplifyAppByJobId = _.groupBy(amplifyApp, (appInfo: AmplifyAppInfo) => {
     if (Object.keys(appInfo.backends).length === 0) {
       return ORPHAN;
     }
 
-    const buildIds = _.groupBy(appInfo.backends, backendInfo => _.get(backendInfo, ['cciInfo', 'id'], UNKNOWN));
+    const buildIds = _.groupBy(appInfo.backends, backendInfo => _.get(backendInfo, ['jobId'], UNKNOWN));
     if (Object.keys(buildIds).length === 1) {
       return Object.keys(buildIds)[0];
     }
 
     return MULTI_JOB_APP;
   });
-
+  const codeBuildJobIds: string[] = _.uniq([...Object.keys(stacksByJobId), ...Object.keys(bucketByJobId), ...Object.keys(amplifyAppByJobId)])
+  .filter((jobId: string) => jobId !== UNKNOWN && jobId !== ORPHAN && jobId !== MULTI_JOB_APP)
+  const buildInfos = await getJobCodeBuildDetails(codeBuildJobIds);
+  const buildInfosByJobId = _.groupBy(buildInfos, (build: CodeBuild.Build) => _.get(build, ['id']));
```

Comment:
This part is to fix the throttling error of calling batchGetBuild for multiple times. Now it only calls once per account